docs/tutorials/dataassociation/KDTree_Tutorial.py

```diff
@@ -0,0 +1,403 @@
+#
+# .. image:: ../../_static/kd_tree_fig_1.png
+#   :width: 1100
+#   :height: 250
+#   :alt: Image showing diagrammatical representation of kd-tree in 2 dimensions, described further in figure 1 caption.
+#
+# Fig. 1a offers a graphical representation of the :math:`k`-d tree shown in 1b. :math:`d` represents the depth of
+# the tree. The :math:`k`-d tree is constructed from a series of detections, which are 2-dimensional vectors, with
+# splits alternating in each dimension: at :math:`d=0` the split occurs in the :math:`x`-axis, at :math:`d=1` the
+# split occurs in the :math:`y`-axis etc. Colours of the splitting lines in 1a correspond with the splits at each
+# depth shown in 1b.
+
+
+# %%
+# Searching a :math:`k`-d tree with Nearest Neighbour
+# ---------------------------------------------------
+# Once the :math:`k`-d tree has been constructed from the detection set, it can then be used to search for the Nearest
+# Neighbour of a given track prediction (which is the query vector).
+# 
+# The method for searching the tree is similar to the method for building the tree: starting at the root node, the
+# :math:`x^{th}` dimension of the query vector is compared with :math:`x^{th}` dimension of the root
+# node vector, if the value of the query vector is less than or equal to the root node detection vector, the search
+# moves down the left subtree, and down the right subtree if the value is greater. This process is repeated recursively
+# at each internal node, again alternating through the vector dimensions, until a leaf node is reached. At each node,
+# the distance between the query vector and the node vector is measured, and if the distance is less than the current
+# best, the node vector becomes the new current best nearest neighbour.
+# 
+# Once the search reaches a leaf node, the search continues back up through the tree to ensure the best nearest
+# neighbour has been identified. From the leaf node, it will traverse back up to the parent node. If the distance from
+# the query vector and the split plane (shown by the blue lines in Fig. 2a) of the parent node is less than the current
+# best, the search will proceed down the sibling subtree as above. If the distance is greater than the current best, the
+# search will proceed up to the next parent node. The search continues this way recursively up the tree until we return
+# to the root node.
+#
+# .. image:: ../../_static/kd_tree_fig_2.png
+#   :width: 1200
+#   :height: 250
+#   :alt: Image showing diagrammatical representation of kd-tree in 2 dimensions, described further in figure 2 caption.
+#
+# Fig. 2a offers a graphical representation of the :math:`k`-d tree shown in 2b. 2a shows the query vector,
+# :math:`[8,2]`, in blue. 2b shows the first part of the search taking place down the :math:`k`-d tree, with the text
+# on the right indicating the direction of the search at each depth. Current best indicates the current best distance of
```

Comment:
How is the distance calculated? Looking at the diagram and the "best distances" it looks to be using the Manhattan metric? Would be useful to specify for clarity.